integrations/sharepoint/src/SharepointClient.ts

```diff
@@ -0,0 +1,441 @@
+      .post(
+        url,
+        {
+          clientState: "A0A354EC-97D4-4D83-9DDB-144077ADB449",
+          resource: `https://${this.primaryDomain}.sharepoint.com/sites/${this.siteName}/_api/web/lists('${listId}')`,
+          notificationUrl: webhookurl,
+          expirationDateTime: new Date(Date.now() + 1000 * 60 * 60 * 24 * 30).toISOString(),
+        },
+        {
+          headers: {
+            Authorization: `Bearer ${token.accessToken}`,
+            Accept: "application/json;odata=verbose",
+          },
+        }
+      )
+      .catch(handleAxiosError);
+
+    if (!res) {
+      throw new sdk.RuntimeError(`Error registering webhook`);
+    }
+
+    const webhookId = res.data.d.id;
+
+    return webhookId;
+  }
+
+  /**
+   * Unregisters a webhook for the configured document library.
+   * @param webhookId - The ID of the webhook to unregister.
+   */
+  async unregisterWebhook(webhookId: string): Promise<void> {
+    const listId = this.getDocumentLibraryListId();
+    const url = `https://${this.primaryDomain}.sharepoint.com/sites/${this.siteName}/_api/web/lists('${listId}')/subscriptions('${webhookId}')`;
+    const token = await this.acquireToken();
+    await axios
+      .delete(url, {
+        headers: {
+          Authorization: `Bearer ${token.accessToken}`,
+          Accept: "application/json;odata=verbose",
+        },
+      })
+      .catch(handleAxiosError);
+  }
+
+  /**
+   * Get the list of documents from the configured document library.
+   * @returns - The list of documents.
+   */
+  private async listItems(): Promise<SharePointItem[]> {
+    const token = await this.acquireToken();
+    const url = `https://${this.primaryDomain}.sharepoint.com/sites/${this.siteName}/_api/web/lists/getbytitle('${this.documentLibraryName}')/items`;
+    const res = await axios.get<SharePointItemsResponse>(url, {
+      headers: {
+        Authorization: `Bearer ${token.accessToken}`,
+        Accept: "application/json;odata=verbose",
+      },
+    });
+    return res.data.d.results;
+  }
+
+  /**
+   * Get the file name from the SharePoint list item
+   * @param listItemIndex - The item index to get the file name for
+   * @returns - The file name or null if the file does not exist
+   */
+  private async getFileName(listItemIndex: number): Promise<string | null> {
+    // sample url -https://botpressio836.sharepoint.com/sites/DemoStandardTeamPage/_api/web/lists/getbytitle('NewDL')/items(3)/File
+    const url = `https://${this.primaryDomain}.sharepoint.com/sites/${this.siteName}/_api/web/lists/getbytitle('${this.documentLibraryName}')/items(${listItemIndex})/File`;
+    const token = await this.acquireToken();
+    const res = await axios
+      .get(url, {
+        headers: {
+          Authorization: `Bearer ${token.accessToken}`,
+          Accept: "application/json;odata=verbose",
+        },
+      })
+      .catch(() => {
+        return {
+          data: {
+            d: {
+              Name: null,
```

Comment:
**question:** does it make sense to throw an error here, given that you’re already handling exceptions where this method is called?

Reply:
Oh good catch, I mean't to remove those try-catch. This is the only error I expect and I wanted to handled it more exclusively than a try catch around more operations. 

Removed the try catch blocks.